File: 1422.分割字符串的最大得分.py
#
# @lc app=leetcode.cn id=1422 lang=python3
#
# [1422] 分割字符串的最大得分
#

# @lc code=start
class Solution:
    def maxScore(self, s: str) -> int:
        sum = s.count('1')
        ans = 0
        for i in s[:-1]:
            sum = sum-1 if i == '1' else sum +1
            ans = max(ans, sum)
        return ans
# @lc code=end

File: 1422.分割字符串的最大得分.cpp
/*
 * @lc app=leetcode.cn id=1422 lang=cpp
 *
 * [1422] 分割字符串的最大得分
 */

// @lc code=start
class Solution {
public:
    int maxScore(string s) {
        int sum = 0;
        int right = count(s.begin(), s.end(), '1');
        int left = 0;
        if(s[0] == '0')
        {
            left++;
            sum = max(sum, left+right);
        }
        else{
            right--;
        }
        for(int i = 1; i < s.size(); i++){
            sum = max(sum, left+right);
            if(s[i] == '0')
            {
                left++;
            }
            else{
                right--;
            }
            // sum = max(tmp, sum);
        }
        return sum;
    }
};
// @lc code=end

